photon-api/src/test/scala/com/linkedin/photon/ml/function/glm/DistributedGLMLossFunctionTest.scala

```diff
@@ -0,0 +1,126 @@
+
+  import DistributedGLMLossFunctionTest._
+
+  /**
+   * Verify the value of loss function without regularization.
+   */
+  @Test()
+  def testValueNoRegularization(): Unit = {
+    val fixedEffectRegularizationContext = NoRegularizationContext
+    val fixedEffectOptimizationConfiguration = FixedEffectOptimizationConfiguration(
+      fixedEffectOptimizerConfig,
+      fixedEffectRegularizationContext)
+    val distributedGLMLossFunction = DistributedGLMLossFunction(
+      fixedEffectOptimizationConfiguration,
+      LogisticLossFunction,
+      treeAggregateDepth)
+    val value = distributedGLMLossFunction.value(
+      labeledPoints,
+      coefficients,
+      PhotonNonBroadcast(normalizationContext))
+
+    // expectValue = log(1 + exp(3)) + log(1 + exp(2)) = 5.1755
+    assertEquals(value, 5.1755, epsilon)
+  }
+
+  /**
+   * Verify the value of loss function with L2 regularization.
+   */
+  @Test()
+  def testValueWithL2Regularization(): Unit = {
+    val fixedEffectRegularizationContext = L2RegularizationContext
+    val fixedEffectOptimizationConfiguration = FixedEffectOptimizationConfiguration(
+      fixedEffectOptimizerConfig,
+      fixedEffectRegularizationContext,
+      fixedEffectRegularizationWeight)
+    val distributedGLMLossFunction = DistributedGLMLossFunction(
+      fixedEffectOptimizationConfiguration,
+      LogisticLossFunction,
+      treeAggregateDepth)
+    val value = distributedGLMLossFunction.value(
+      labeledPoints,
+      coefficients,
+      PhotonNonBroadcast(normalizationContext))
+
+    // expectedValue = log(1 + exp(3)) + log(1 + exp(2)) + 1 * ((-2)^2 + 3^2) / 2 = 11.6755
+    assertEquals(value, 11.6755, epsilon)
+  }
+
+  /**
+   * Verify the value of loss function with elastic net regularization.
+   */
+  @Test()
+  def testValueWithElasticNetRegularization(): Unit = {
+    val fixedEffectRegularizationContext = ElasticNetRegularizationContext(alpha)
+    val fixedEffectOptimizationConfiguration = FixedEffectOptimizationConfiguration(
+      fixedEffectOptimizerConfig,
+      fixedEffectRegularizationContext,
+      fixedEffectRegularizationWeight)
+    val distributedGLMLossFunction = DistributedGLMLossFunction(
+      fixedEffectOptimizationConfiguration,
+      LogisticLossFunction,
+      treeAggregateDepth)
+    val value = distributedGLMLossFunction.value(
+      labeledPoints,
+      coefficients,
+      PhotonNonBroadcast(normalizationContext))
+
+    // L1 is computed by the optimizer.
+    // expectedValue = log(1 + exp(3)) + log(1 + exp(2)) + (1 - 0.4) * 1 * ((-2)^2 + 3^2) / 2 = 9.0755
+    assertEquals(value, 9.0755, epsilon)
+  }
+}
+
+object DistributedGLMLossFunctionTest {
+  private val fixedEffectOptimizerType = OptimizerType.LBFGS
+  private val fixedEffectMaxIter = 1
+  private val fixedEffectTolerance = 1.0
+  private val fixedEffectOptimizerConfig = OptimizerConfig(
+    fixedEffectOptimizerType,
+    fixedEffectMaxIter,
+    fixedEffectTolerance)
```

Comment:
Lines 107 - 113 can be replaced by a mock `OptimizerConfig`. It should work immediately since no mock object functions will be called during the test.

Reply:
Done